Sprint-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,62 @@
 const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+  "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-d.jpg",
 ];
 
+// Write your code here
+const carouselImg = document.getElementById("carousel-img");
+const prevBtn = document.getElementById("backward-btn");
+const nextBtn = document.getElementById("forward-btn");
+const autoForwardBtn = document.getElementById("auto-forward-btn");
+const stopBtn = document.getElementById("stop-btn");
+const autoBackwardBtn = document.getElementById("auto-backward-btn");
 
-// Write your code here
\ No newline at end of file
+let currentImageIndex = 0;
+
+const updatePicture = () => {
+  //make pics the same size
+  carouselImg.style.width = "400px";
+  carouselImg.style.height = "400px";
+  carouselImg.style.objectFit = "contain";
+  //swap to next
+  carouselImg.src = images[currentImageIndex];
+};
+
+const back = () => {
+  if (currentImageIndex === 0) {
+    currentImageIndex = images.length - 1;
+  } else {
+    currentImageIndex = currentImageIndex - 1;
+  }
+};
+//add event listeners to buttons
+prevBtn.addEventListener("click", () => {
+  back();
+  updatePicture();
+});
+
+const forward = () => {
+  if (currentImageIndex === images.length - 1) {
+    currentImageIndex = 0;
+  } else {
+    currentImageIndex = currentImageIndex + 1;
+  }
+};
+
+nextBtn.addEventListener("click", () => {
+  forward();
+  updatePicture();
+});
+
+//automated slideshow button event listeners
+autoForwardBtn.addEventListener("click", () => setInterval(forward, 1500));
```

Comment:
This would probably work pretty well, but our `forward` function here just changes the index of the picture that _would_ be shown if `updatePicture` were called. The idea of separately updating the picture is a pretty good one, and sometimes that pattern could be what you want, but maybe in this case it might be better to call `updatePicture` at the end of the `forward` and `back` functions? It seems that every time we call `forward`, for example, we're going to want to change the display of the image. If things were to get more complicated then maybe you'd want to call the `updatePicture` function separately.